### PR TITLE
Fix Dirloader to work with acquired dirs (-ot)

### PR DESCRIPTION
**Pull request by cecinestpasunepipe**
(DIS-1856)

### CODE REVIEW

File: dissect/target/loaders/dir.py [hunk 1]
```diff
@@ -17,9 +17,13 @@ class DirLoader(Loader):
 
     @staticmethod
     def detect(path: Path) -> bool:
+        if (path / "fs").exists():
```
Comment (Schamper) [inline dissect/target/loaders/dir.py:20]: I think this check is a bit too assuming. Could you add some more conditions? E.g. only do this if `find_dirs` returns `None` first?

File: dissect/target/loaders/dir.py [hunk 2]
```diff
@@ -15,15 +16,24 @@
...
+
+    @classmethod
+    @cache
+    def _find_entry_path(cls, path: Path):
+        found = None
+        for prefix in cls.PREFIXES:
+            if find_dirs(path / prefix)[0] is not None:
+                found = prefix
+                break
+        return found
```
Comment (Schamper) [inline dissect/target/loaders/dir.py:29]:
```suggestion
        for prefix in cls.PREFIXES:
            if find_dirs(path / prefix)[0] is not None:
                return prefix
```

Comment (cecinestpasunepipe) [reply dissect/target/loaders/dir.py]: I always try to avoid multiple returns in one function, but this is more compact. Trade-off I guess.

**Comment by cecinestpasunepipe** [comment]
Unfortunately the prefix-function can't be cached due to side-effects.

**Review by Horofic — DISMISSED** [review]

File: dissect/target/loaders/dir.py [hunk 3]
```diff
@@ -15,11 +15,20 @@
 class DirLoader(Loader):
     """Load a directory as a filesystem."""
 
+    PREFIXES = ["", "fs"]
+
+    @classmethod
+    def _find_entry_path(cls, path: Path) -> str | None:
```
Comment (Schamper) [inline dissect/target/loaders/dir.py:21]: Maybe you can make this a function in the global scope, similar to the other functions here. There's not much reason for it to be a classmethod, is there?

File: dissect/target/loaders/dir.py [hunk 4]
```diff
@@ -11,15 +11,24 @@
 if TYPE_CHECKING:
     from dissect.target import Target
 
+PREFIXES = ["", "fs"]
+
+
+def find_entry_path(path: Path) -> str | None:
```
Comment (Schamper) [inline dissect/target/loaders/dir.py:17]:
```suggestion
def find_entry_path(path: Path) -> Optional[str]:
```
I don't think using `|` in type hints is legal in Python 3.9?

Could you also move this below the `DirLoader` class itself? That way it's more consistent placing with the other functions.

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — CHANGES_REQUESTED** [review]

**Review by Schamper — APPROVED** [review]

**Pull request merged**